components/board/genericlinux/gpio.go

```diff
@@ -122,6 +123,19 @@ func (pin *gpioPin) Set(ctx context.Context, isHigh bool,
 // This function assumes you've already locked the mutex. It sets the value of a pin without
 // changing whether the pin is part of a software PWM loop.
 func (pin *gpioPin) setInternal(isHigh bool) (err error) {
+	// if pin is a hw pwm pin, set to either 0% or 100% duty cycle
+	if pin.hwPwm != nil {
+		var value float64
+		if isHigh {
+			value = 1
+		}
+		err := pin.hwPwm.SetPwm(10, value)
```

Comment:
What's the 10 on this line? I think it's the frequency, and then I wonder why we're changing that from whatever the old frequency was. I suspect the answer is "if the old frequency is unset, we need to make it some legitimate value, and 10 is good enough. If the old frequency was already set, it's okay to change it to some other value here, and we'll change it back if someone sets it again later. Note that we're not modifying `pin.pwmFreqHz`, so the old value is still available for later use." I'd put in a comment like that in the code, but mine is too wordy.

Reply:
will add a comment about this!

Reply:
Is there any specific reason why the frequency is set to 10? I guess I'm wondering if it should be lower/higher for performance reasons / to properly recreate GPIO. I'm assuming the difference in frequency is negligible but wanted to ask anyways!